usage-rules.md

```diff
@@ -0,0 +1,237 @@
+# Usage Rules for Mimic Library
+
+This document provides essential guidelines for coding agents when using the Mimic mocking library in Elixir projects.
```

Comment:
Maybe we could add something about not mocking standard library modules? Like DateTime, Integer, `:ets`, etc?